src/eyreish/wrapper.rs

```diff
@@ -6,35 +6,9 @@ use crate::{Diagnostic, LabeledSpan, Report, SourceCode};
 
 use crate as miette;
 
-#[repr(transparent)]
-pub(crate) struct DisplayError<M>(pub(crate) M);
-
 #[repr(transparent)]
 pub(crate) struct MessageError<M>(pub(crate) M);
 
-pub(crate) struct NoneError;
-
-impl<M> Debug for DisplayError<M>
-where
-    M: Display,
-{
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        Display::fmt(&self.0, f)
-    }
-}
-
-impl<M> Display for DisplayError<M>
-where
-    M: Display,
-{
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        Display::fmt(&self.0, f)
-    }
-}
-
-impl<M> StdError for DisplayError<M> where M: Display + 'static {}
-impl<M> Diagnostic for DisplayError<M> where M: Display + 'static {}
-
```

Comment:
wasn't all this here for a reason? I'm nervous about this change cause it touches the code that was taken from anyhow, which I honestly don't fully understand, and does some Weird Things with pointers and types.

Reply:
I don't know either but I couldn't find any mention of `DisplayError` in the whole Miette history except for adding the struct and its impl. It is not public outside of the crate either (and thus the Miri error), unlike the one in `anyhow`. So I don't know how/where it could be useful and figured deleting it was the best option.

If you don't agree, I can easily change that to `dead_code` instead.
(and if so, does your comment apply to `NoneError` as well? I couldn't find it in `anyhow`'s code, so I'm not sure if it does)

Reply:
Additional info: if I understand correctly, `DisplayError` in Anyhow (and Eyre, which also has a `DisplayError` but private to the crate, like Miette and unlike Anyhow, where its public) is used to convert an `Option<T>` to a `Result<T>`. Miette does not have such a feature.

Reply:
2 more points, in Eyre:
- in Eyre, `DisplayError` seems to only be used for compatibility with Anyhow (via `ContextCompat`). For the pure Eyre (`ok_or_eyre`), the `Option`->`Result` uses `MessageError` instead (which is also not available in Miette).
- to answer my own question about `NoneError`, it is in Eyre and used along side `DisplayError`, for the same Option-to-Result in that `ContextCompat`.

So at this point, I'm 99% sure that it's safe to remove. And I believe I can remove an extra line of code related to that Option conversion.